File: Lesson09_OOP/05_Encapsulation/Shirt.py
"""
Copyright Khawla Shnaikat and ChatGPT 3.5, 2023
Licensed under GPL v3
See LICENSE.txt for more information.
"""
from clothing import Clothing
class Shirt:
    SLEEVETYPES = ["short", "three-quarter", "long"]

    def __init__(self, colour, material, sleeve):
        self.fabric = Clothing(colour, material)
        self.set_sleeve_type(sleeve)

    def get_material(self):
        return self.fabric.get_material()

    def get_colour(self):
        return self.fabric.get_colour()

    def get_colour_id(self):
        return self.fabric.get_colour_id()

    def get_sleeve_type(self):
        return self.SLEEVETYPES[self.sleeve_type]

    def set_material(self, material):
        self.fabric.set_material(material)

    def set_colour(self, colour):
        self.fabric.set_colour(colour)

    def set_sleeve_type(self, sleeve):
        self.sleeve_type = sleeve if sleeve < len(self.SLEEVETYPES) else 0
if __name__ == "__main__":
    my_shirt = Shirt("white", "cotton", 2)
    print(f"Shirt Colour: {my_shirt.get_colour()}, Material: {my_shirt.get_material()}, Sleeve: {my_shirt.get_sleeve_type()}")

File: Lesson09_OOP/05_Encapsulation/Shirt.java
/*
Copyright Ann Barcomb and Emily Marasco, 2022
Licensed under GPL v3
See LICENSE.txt for more information.
*/

public class Shirt {
    private static final String SLEEVETYPES[] = {"short", "three-quarter", "long"};
    private Clothing fabric;
    private int sleeveType;

    public Shirt(String colour, String material, int sleeve) {
        this.fabric = new Clothing(colour, material);
        this.setSleeveType(sleeve);
    }

    public Shirt(int colour, String material, int sleeve) {
        this.fabric = new Clothing(colour, material);
        this.setSleeveType(sleeve);
    }

    public String getMaterial() { return this.fabric.getMaterial(); }
    public String getColour() { return this.fabric.getColour(); }
    public int getColourID() { return this.fabric.getColourID(); }
    public String getSleeveType() { return SLEEVETYPES[this.sleeveType]; }
    public void setMaterial(String material) { this.fabric.setMaterial(material); }
    public void setColour(String colour) { this.fabric.setColour(colour); }
    public void setColour(int colour) { this.fabric.setColour(colour); }
    public void setSleeveType(int sleeve) {
        if (sleeve < SLEEVETYPES.length) {
            this.sleeveType = sleeve;
        } else {
            this.sleeveType = 0;
        }
    }

    public static void main(String[] args) {
        var myShirt = new Shirt(2, "cotton", 2);
        System.out.printf("Colour: %s, Material: %s, Sleeve: %s\n", 
          myShirt.getColour(), myShirt.getMaterial(), myShirt.getSleeveType());
    }
}


